### PR TITLE
Add new repositories for the backend checks lambdas/libraries.

### PR DESCRIPTION
**Pull request by MancunianSam**
I've also tidied up the terraform policy. It's getting too large again
so I've added some *s in.

### CODE REVIEW

File: modules/environment-roles/templates/shared_iam_terraform_policy.json.tpl [hunk 1]
```diff
@@ -31,10 +31,9 @@
         "arn:aws:iam::${account_id}:policy/TDRCreateDbUsersConsignmentapiPolicy${title(environment)}",
         "arn:aws:iam::${account_id}:policy/TDRCreateKeycloak*",
         "arn:aws:iam::${account_id}:policy/TDRCustodian*",
-        "arn:aws:iam::${account_id}:policy/TDRDbMigrationLambdaPolicy${title(environment)}",
+        "arn:aws:iam::${account_id}:policy/*Lambda*${title(environment)}",
```
Comment (TomJKing) [inline modules/environment-roles/templates/shared_iam_terraform_policy.json.tpl:34]: Could we keep the `TDR`, just to control the access a bit more, eg: `.../TDR*Lambda*...`

File: modules/environment-roles/templates/shared_iam_terraform_policy.json.tpl [hunk 2]
```diff
@@ -77,27 +72,22 @@
         "arn:aws:iam::${account_id}:role/TDRCreateDbUsersBastionRole${title(environment)}",
         "arn:aws:iam::${account_id}:role/TDRCreateDbUsersConsignmentapiRole${title(environment)}",
         "arn:aws:iam::${account_id}:role/TDRCreateKeycloak*",
-        "arn:aws:iam::${account_id}:role/TDRDbMigrationLambdaRole${title(environment)}",
+        "arn:aws:iam::${account_id}:role/*Lambda*${title(environment)}",
```
Comment (TomJKing) [inline modules/environment-roles/templates/shared_iam_terraform_policy.json.tpl:75]: Could we keep the `TDR`, just to control the access a bit more, eg: `.../TDR*Lambda*...`

File: root_github.tf [hunk 3]
```diff
@@ -631,3 +631,46 @@ module "github_file_upload_data_repository" {
     WORKFLOW_PAT       = module.common_ssm_parameters.params[local.github_access_token_name].value
   }
 }
+
+module "github_backend_checks_utils_environment" {
```
Comment (TomJKing) [inline root_github.tf:635]: Should this resource be called `github_backend_checks_utils_repository`  to keep it consistent with the other resource names?

**Review by TomJKing — COMMENTED** [review]
Looks good. Couple of minor comments.

File: modules/environment-roles/templates/shared_iam_terraform_policy.json.tpl (hunk 1 above)
Comment (MancunianSam) [reply modules/environment-roles/templates/shared_iam_terraform_policy.json.tpl:34]: Yeah we can, I checked the new roles and policies and they are called TDRxxx. I've changed it and the same for the roles below.

File: root_github.tf (hunk 3 above)
Comment (MancunianSam) [reply root_github.tf:635]: Yep, must have copied over from terraform-environments :smile: I've changed it.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**